src/invstorage.cc

```diff
@@ -52,9 +52,63 @@ void INvStorage::importData(const ImportConfig& import_config) {
 boost::shared_ptr<INvStorage> INvStorage::newStorage(const StorageConfig& config) {
   switch (config.type) {
     case kSqlite:
+      if (!boost::filesystem::exists(config.sqldb_path)) {
+        boost::shared_ptr<INvStorage> sql_storage(new SQLStorage(config));
+        boost::shared_ptr<INvStorage> fs_storage(new FSStorage(config));
+        INvStorage::FSSToSQLS(fs_storage, sql_storage);
```

Comment:
It seems to me that it will only work for the first time, and on subsequent calls when the FS storage has been cleared it will just populate our SQLStorage with empty values. There is also a minor problem with FSStorage constructor creating directories which we don't want to be there once we switched to SQL storage.

Reply:
No, the second time we will not run the migration because we check if database exists. SQLStorage constructor will create the databse at first time. Also we will not run FSStorage constructor second time because it runs only if database not exists.

Reply:
Thank you, makes sense.